File: Maximum_Ice_Cream_Bars.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	costs := []int{1, 3, 2, 4, 1}
	coins := 7
	fmt.Println(maxIceCream(costs, coins))
}

func maxIceCream(costs []int, coins int) int {
	sort.Ints(costs)
	res := 0
	for _, v := range costs {
		if coins >= v {
			coins -= v
			res += 1
		}
	}
	return res
}

File: Maximum_Ice_Cream_Bars/Maximum_Ice_Cream_Bars.py
costs = [1,3,2,4,1]
coins = 7

def maxIceCream(costs,coins):
    costs.sort()

    res = 0
    for i in range(len(costs)):
        if coins >= costs[i]:
            coins -= costs[i]
            res +=1
    
    return res

print(maxIceCream(costs,coins))